agol_connection.py

```diff
@@ -0,0 +1,119 @@
+from arcgis.gis import GIS
+from arcgis.features import FeatureLayerCollection, FeatureSet, Table, Feature
+import tempfile
+import shutil
+import json
+import csv
+import os
+
+
+class AGOLConnection(object):
+
+    def __init__(self):
```

Comment:
(not for this PR, maybe, but as a side note we should PEP-8 all this code sometime)